tapdb/burn_tree.go

```diff
@@ -204,221 +157,189 @@ func (bt *BurnUniverseTree) InsertBurns(ctx context.Context,
 	return lfn.Ok(finalResults)
 }
 
-// QueryBurns attempts to query a set of burn leaves for the given asset
-// specifier. If the burn leaf points are empty, then all burn leaves are
-// returned.
-func (bt *BurnUniverseTree) QueryBurns(ctx context.Context,
+// queryBurnLeaves retrieves UniverseLeaf records based on burn OutPoints.
+func queryBurnLeaves(ctx context.Context, dbtx BaseUniverseStore,
 	spec asset.Specifier,
-	burnPoints ...wire.OutPoint) universe.BurnLeafQueryResp {
+	burnPoints ...wire.OutPoint) ([]UniverseLeaf, error) {
 
-	// Derive identifier from the asset.Specifier.
-	id, err := burnSpecifierToIdentifier(spec)
+	uniNamespace, err := specifierToIdentifier(
+		spec, universe.ProofTypeBurn,
+	)
 	if err != nil {
-		return lfn.Err[lfn.Option[[]*universe.AuthenticatedBurnLeaf]](
-			err,
-		)
+		return nil, fmt.Errorf("error deriving identifier: %w", err)
 	}
-	namespace := id.String()
 
-	var resultLeaves []*universe.AuthenticatedBurnLeaf
+	var leavesToQuery []UniverseLeaf
 
-	readTx := NewBaseUniverseReadTx()
-	txErr := bt.db.ExecTx(ctx, &readTx, func(db BaseUniverseStore) error {
-		// Create the tree within the transaction for getting root and
-		// proofs.
-		tree := mssmt.NewCompactedTree(
-			newTreeStoreWrapperTx(db, namespace),
-		)
-
-		// Get the tree root once for all queries. Handle the case
-		// where the tree might be empty.
-		root, err := tree.Root(ctx)
-		if err != nil {
-			return err
-		}
+	namespace := uniNamespace.String()
 
-		var leavesToQuery []UniverseLeaf
-
-		switch {
-		// If specific burn points are provided, query for each of them.
-		case len(burnPoints) > 0:
-			for _, burnPoint := range burnPoints {
-				burnPointBytes, err := encodeOutpoint(burnPoint)
-				if err != nil {
-					return fmt.Errorf("unable to encode "+
-						"burn point %v: %w", burnPoint,
-						err)
-				}
-
-				// Query leaves matching the burn point and
-				// namespace. ScriptKeyBytes is nil here as we
-				// want all script keys for this burn point.
-				dbLeaves, err := db.QueryUniverseLeaves(
-					ctx, UniverseLeafQuery{
-						MintingPointBytes: burnPointBytes, //nolint:lll
-						Namespace:         namespace,
-					},
-				)
-				if err != nil {
-					// If no rows found for a specific
-					// point, continue to the next.
-					if errors.Is(err, sql.ErrNoRows) {
-						continue
-					}
-					return fmt.Errorf("error querying "+
-						"leaves for burn point %v: %w",
-						burnPoint, err)
-				}
-
-				if len(dbLeaves) == 0 {
-					continue
-				}
-
-				leavesToQuery = append(
-					leavesToQuery, dbLeaves...,
-				)
+	// If specific burn points are provided, query for each of them.
+	if len(burnPoints) > 0 {
```

Comment:
nit: since we're in a helper function now, we could invert the condition of this `if`, do the general `QueryUniverseLeaves` query and return. Then we could do this block un-indented further down, saving a whole indentation-level for the nested loop.